### PR TITLE
Ctw 353/update logic in condition display

### PR DESCRIPTION
**Pull request by luthfib**
This pr adds more codes to filter out duplicate conditions

TODO: 
- [x] Fix type issue in fhir/conditions.ts
- [x] Find more conditions to test or create conditions to test
- [x] Changeset

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -7,7 +7,7 @@ const { VITE_SYSTEM_URL, VITE_AUTH_TOKEN, VITE_PATIENT_ID } = import.meta.env;
 
 function App() {
   return (
-    <CTWProvider env="dev" authToken={VITE_AUTH_TOKEN} theme={{}}>
+    <CTWProvider env="sandbox" authToken={VITE_AUTH_TOKEN} theme={{}}>
```
Comment (pmadria3) [inline src/App.tsx:10]: probably should use dev here

File: src/fhir/conditions.ts [hunk 2]
```diff
@@ -98,3 +108,38 @@ function filterAndSort(conditions: fhir4.Condition[]) {
...
+
+  data.forEach((condition) => {
+    const conditionModel = new ConditionModel(condition);
+    ACCEPTABLE_CODES.forEach((code) => {
+      if (!(code in confirmedCodeDict)) {
+        confirmedCodeDict[code] = [];
+      }
+      if (typeof conditionModel[code] === "string") {
+        const modelCode = conditionModel[code];
+        confirmedCodeDict[code].push(modelCode as string);
```
Comment (pmadria3) [inline src/fhir/conditions.ts:123]: nit: can just use conditionModel[code] directly instead of setting to modelCode

File: src/fhir/conditions.ts [hunk 3]
```diff
@@ -98,3 +108,38 @@ function filterAndSort(conditions: fhir4.Condition[]) {
...
+export const filterDuplicateCodesFromTarget = (
+  target: fhir4.Condition[],
+  codesLookup: ConfirmedCodes
+) =>
+  target.filter((c) => {
+    const isDuplicate = ACCEPTABLE_CODES.some((code) => {
+      const conditionModel = new ConditionModel(c);
+      if (conditionModel[code]) {
+        const modelCode = conditionModel[code];
+        return codesLookup[code].includes(modelCode as string);
```
Comment (pmadria3) [inline src/fhir/conditions.ts:140]: same nit as above

**Review by pmadria3 — DISMISSED** [review]
Small nits but otherwise LGTM

**Review by pmadria3 — APPROVED** [review]
LGTM

**Review by WesleyKapow — CHANGES_REQUESTED** [review]
Quick feedback -- It'd be useful if we could have an array of SYSTEM URLs of systems to use. Where all we need to do is edit this array to edit how things work. I think this means moving away from having explicit `icd9Code` and `icd10CMCode` getters on the model.

We're going to use this array in a few places. For matching and removing. And also for history lookup.

**Comment by luthfib** [comment]
That makes sense. That's where my thought was with the codes array but an array of system urls would simplify it even more as we wouldnt have to add methods to the condition model

File: src/models/conditions.ts [hunk 4]
```diff
@@ -71,6 +73,14 @@ export class ConditionModel {
     );
   }
 
+  get availableCodes(): fhir4.Coding[] {
```
Comment (pmadria3) [inline src/models/conditions.ts:76]: change availableCodes into `allCodeSystems` or `allSystemCodes`?

File: src/models/conditions.ts [hunk 5]
```diff
@@ -78,16 +73,12 @@ export class ConditionModel {
...
-    return findCoding(SYSTEM_ICD9, this.resource.code)?.code;
-  }
-
-  get icd9CMCode(): string | undefined {
-    return findCoding(SYSTEM_ICD9_CM, this.resource.code)?.code;
-  }
-
-  get icd10CMCode(): string | undefined {
-    return findCoding(SYSTEM_ICD10_CM, this.resource.code)?.code;
+  get allSystemCodings(): fhir4.Coding[] {
```
Comment (WesleyKapow) [inline src/models/conditions.ts:76]: How about `knownCodings`?

**Review by WesleyKapow — APPROVED** [review]
Handful of naming nitpicks to make this a bit more clear, otherwise LGTM

**Review by WesleyKapow — APPROVED** [review]

**Pull request merged**